moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -369,10 +373,21 @@ Eigen::VectorXd Servo::jointDeltaFromCommand(const ServoInput& command, const mo
     {
       try
       {
-        const TwistCommand command_in_planning_frame = toPlanningFrame(std::get<TwistCommand>(command));
-        delta_result =
-            jointDeltaFromTwist(command_in_planning_frame, robot_state, servo_params_, joint_name_group_index_map);
-        servo_status_ = delta_result.first;
+        const auto ik_solver = active_joint_model_group->getSolverInstance();
+        if (ik_solver)
+        {
+          const auto planning_frame = ik_solver->getBaseFrame();
```

Comment:
Is there a better way to get the base link of a joint model group without having to grab the IK solver like this?

Reply:
Can't you do this?

```
const JointModel* root_joint_model = group->getJointModels().front();
const LinkModel* root_link_model = root_joint_model->getParentLinkModel();
```

But now I wonder if you rather need the IK solver base frame (i.e. your code as is), since the underlying IK solver could be working on a different frame, not necessarily the group root?
If that's the case, maybe create a `getIKSolverBaseFrame()` function that can be called here and below?

Reply:
Thanks!

This raises a follow up question that, if these changes are the right thing for changing subgroups, can we actually get rid of the planning frame parameter and have servo always do its own internal conversions to active group's IK base frame?

Reply:
Hmm.. looking deeper into Servo I got more confused about what the 'planning frame' is, but also how this fix works, since the IK base frame for the manipulator group is the one tilted 45 deg. right? so I would expect this to still move the wrong way.
Another q: the command has a `frame_id`. Shouldn't that be all that is needed to move in the same frame (e.g. `world`), independently of the active group? Sorry I have more questions than answers!

Reply:
You should have more questions than answers, I think there's a lot here that needs to be disentangled for clarity.

Basically, Servo has this rule that enforces that all commands passed into it are expressed w.r.t. a common frame -- this is the "planning frame" we speak of.

All we're trying to do, perhaps in an overly complex way, is the following:

* Receive a Cartesian command w.r.t. some arbitrary frame ID
* Convert it to a reference frame that agrees with the IK solver (this is our planning frame)
* ???
* Profit

Reply:
Agreed, as I was thinking about this today as well.

I think this parameter was carryover from before and we can probably remove it if this is how we're going to handle things going forward.

Reply:
We can probably also get rid of the `ee_frame` parameter if that's the case... given these descriptions:

```
  planning_frame: {
    type: string,
    description: "The name of moveit planning frame.\
                  This parameter does not have a default value and \
                  must be passed to the node during launch time."
  }

  ee_frame: {
    type: string,
    description: "The name of end effector frame of your robot, \
                  this should also be the IK tip frame of your IK solver. \
                  This parameter does not have a default value and \
                  must be passed to the node during launch time."
  }
```

Reply:
I think the answer is, I didn't know about retrieving the IK solver base frame when this was coded 6-years-or-so ago. Automating it sounds nice.

BTW please tag myself and @ibrahiminfinite on Servo PR's

Reply:
Glad to have your vote of confidence to help automate this, I wanted to ask you but didn't want to bug you guys too much with more servo issues until this was out of draft and in a direction we were sure we wanted to take.

Reply:
I remember having some confusing discussions about these frames during the rewrite.
Also agree on the direction of removing the `planning_frame` and `ee_frame` parameters and fetching the right **planning frame**   internally in Servo. 
We already have the frame information REQUIRED from the user in the `frame_id` attribute of the command.

Reply:
Was able to remove them!

Sadly on MoveIt Studio we rely on `moveit_params.ee_frame` to send a command... but that's our problem to go figure out now :)

Reply:
Hmm... I just saw some other piece of code that reminded me of the fact that a planning group can have multiple tip frames. So can we really get rid of `moveit_params.ee_frame` after all?

Reply:
I think we can, at least the way it is now, because `ee_frame` doesn't seem to be used anywhere. The IK computation is done for `ik_solver->getTipFrame()`, so it will always use the first tip.
Longer term what about having Servo expose a `/configure` service, where the user could set the planning group and tip frame?

Reply:
Given all the struggles we've had trying to make sure that parameters stay consistent w.r.t. each other, I quite like the idea of dedicated services that touch these interrelated parameters like planning group + ee frame.

Also, having this `configure()` service means that configuring can implicitly do the task of pausing servo, changing values, and then unpausing it.

Also, I guess to be fair this code was calling `getTipFrame()` already, and also there is an error logged when you call `getTipFrame()` on a planning group with multiple tips.

So maybe this is just a known limitation for now and we proceed as is?

Reply:
A service to update the parameters might be a good idea b/c it would fix this issue:  https://github.com/ros-planning/moveit2/issues/2412

i.e. do the parameter validation in a callback in a different thread

Hate to change the API so drastically, so soon, though. Have to think carefully about this.

Reply:
I think servo has changed enough with the refactor since we kept running into limitations trying to keep the API as stable as possible with the old version. It might be beneficial to do all the big changes as soon as possible in hopes that we get to a somewhat more stable API before people actually start using this new version.

That said, we should try do all this in a way that minimizes our need to keep twiddling the interfaces going forward.